tools/distributions-scraper/main.py

```diff
@@ -0,0 +1,81 @@
+import importlib
+import pkgutil
+import asyncio
+import json
+import pathlib
+import sys
+from concurrent.futures import ThreadPoolExecutor
+from scraper.base import BaseScraper
+import scraper
+
+
+OUTPUT_FILE = (
+    pathlib.Path(__file__).resolve().parent.parent.parent
+    / "data"
+    / "distributions"
+    / "distribution-info.json"
+)
+
+
+def load_scrapers():
+    """Dynamically load all scraper classes from scraper/ package."""
+    scrapers = []
+    for _, module_name, _ in pkgutil.iter_modules(scraper.__path__):
+        if module_name == "base":
+            continue
+
+        module = importlib.import_module(f"scraper.{module_name}")
+        for attr_name in dir(module):
+            obj = getattr(module, attr_name)
+            if isinstance(obj, type) and issubclass(obj, BaseScraper) and obj is not BaseScraper:
+                scrapers.append(obj())
+    return scrapers
+
+
+async def run_scraper(scraper_instance: BaseScraper, executor: ThreadPoolExecutor):
+    loop = asyncio.get_event_loop()
+    name = scraper_instance.name
+    try:
+        result = await loop.run_in_executor(executor, scraper_instance.fetch)
+        return name, result, None
+    except Exception as e:
+        return name, None, str(e)
+
+
+async def main():
```

Comment:
And here, if you can. Places where section comments about what it's doing are good candidates for extraction.